samples/sample_encode/src/pipeline_encode.cpp

```diff
@@ -334,88 +334,101 @@ mfxStatus sTask::Reset()
-    m_MVCSeqDesc.View[1].NumAnchorRefsL0 = 1;
-    m_MVCSeqDesc.View[1].AnchorRefL0[0] = 0;     // ViewId 0 - base view
+    mvcBuffer->View[1].NumAnchorRefsL0 = 1;
+    mvcBuffer->View[1].AnchorRefL0[0] = 0;     // ViewId 0 - base view
 
-    m_MVCSeqDesc.View[1].NumNonAnchorRefsL0 = 1;
-    m_MVCSeqDesc.View[1].NonAnchorRefL0[0] = 0;  // ViewId 0 - base view
+    mvcBuffer->View[1].NumNonAnchorRefsL0 = 1;
+    mvcBuffer->View[1].NonAnchorRefL0[0] = 0;  // ViewId 0 - base view
 
     // viewId array
-    m_MVCSeqDesc.NumViewId = m_nNumView;
-    m_MVCSeqDesc.NumViewIdAlloc = m_nNumView;
-    m_MVCSeqDesc.ViewId = new mfxU16[m_MVCSeqDesc.NumViewIdAlloc];
-    MSDK_CHECK_POINTER(m_MVCSeqDesc.ViewId, MFX_ERR_MEMORY_ALLOC);
-    for (i = 0; i < m_MVCSeqDesc.NumViewIdAlloc; ++i)
+    mvcBuffer->NumViewId = m_nNumView;
+    mvcBuffer->NumViewIdAlloc = m_nNumView;
+    mvcBuffer->ViewId = new mfxU16[mvcBuffer->NumViewIdAlloc];
+    MSDK_CHECK_POINTER(mvcBuffer->ViewId, MFX_ERR_MEMORY_ALLOC);
+    for (i = 0; i < mvcBuffer->NumViewIdAlloc; ++i)
     {
-        m_MVCSeqDesc.ViewId[i] = (mfxU16) i;
+        mvcBuffer->ViewId[i] = (mfxU16) i;
     }
 
     // create a single operation point containing all views
-    m_MVCSeqDesc.NumOP = 1;
-    m_MVCSeqDesc.NumOPAlloc = 1;
-    m_MVCSeqDesc.OP = new mfxMVCOperationPoint[m_MVCSeqDesc.NumOPAlloc];
-    MSDK_CHECK_POINTER(m_MVCSeqDesc.OP, MFX_ERR_MEMORY_ALLOC);
-    for (i = 0; i < m_MVCSeqDesc.NumOPAlloc; ++i)
+    mvcBuffer->NumOP = 1;
+    mvcBuffer->NumOPAlloc = 1;
+    mvcBuffer->OP = new mfxMVCOperationPoint[mvcBuffer->NumOPAlloc];
+    MSDK_CHECK_POINTER(mvcBuffer->OP, MFX_ERR_MEMORY_ALLOC);
+    for (i = 0; i < mvcBuffer->NumOPAlloc; ++i)
     {
-        MSDK_ZERO_MEMORY(m_MVCSeqDesc.OP[i]);
-        m_MVCSeqDesc.OP[i].NumViews = (mfxU16) m_nNumView;
-        m_MVCSeqDesc.OP[i].NumTargetViews = (mfxU16) m_nNumView;
-        m_MVCSeqDesc.OP[i].TargetViewId = m_MVCSeqDesc.ViewId; // points to mfxExtMVCSeqDesc::ViewId
+        MSDK_ZERO_MEMORY(mvcBuffer->OP[i]);
+        mvcBuffer->OP[i].NumViews = (mfxU16) m_nNumView;
+        mvcBuffer->OP[i].NumTargetViews = (mfxU16) m_nNumView;
+        mvcBuffer->OP[i].TargetViewId = mvcBuffer->ViewId; // points to mfxExtMVCSeqDesc::ViewId
     }
 
     return MFX_ERR_NONE;
 }
 
-mfxStatus CEncodingPipeline::AllocAndInitVppDoNotUse()
+mfxStatus CEncodingPipeline::InitVppFilters()
 {
-    m_VppDoNotUse.NumAlg = 4;
+    auto vppExtParams = m_mfxVppParams.AddExtBuffer<mfxExtVPPDoNotUse>();
+
+    vppExtParams->NumAlg = 4;
 
-    if(m_VppDoNotUse.AlgList)
-        delete[] m_VppDoNotUse.AlgList;
-    m_VppDoNotUse.AlgList = new mfxU32 [m_VppDoNotUse.NumAlg];
-    MSDK_CHECK_POINTER(m_VppDoNotUse.AlgList,  MFX_ERR_MEMORY_ALLOC);
+    if(vppExtParams->AlgList)
+        delete[] vppExtParams->AlgList;
+    vppExtParams->AlgList = new mfxU32 [vppExtParams->NumAlg];
+    MSDK_CHECK_POINTER(vppExtParams->AlgList,  MFX_ERR_MEMORY_ALLOC);
```

Comment:
MSDK_CHECK_POINTER not needed here. new throws an exception

Reply:
Removed